docker/tests/containerTestCommon.psm1

```diff
@@ -0,0 +1,202 @@
+$script:forcePull = $true
+# Get docker Engine OS
+function Get-DockerEngineOs
+{
+    docker info --format '{{ .OperatingSystem }}'
+}
+
+# Call Docker with appropriate result checksfunction Invoke-Docker 
+function Invoke-Docker
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string[]]
+        $Command,
+        [ValidateSet("error","warning",'ignore')]
+        $FailureAction = 'error',
+
+        [Parameter(Mandatory=$true)]
+        [string[]]
+        $Params,
+
+        [switch]
+        $PassThru,
+        [switch]
+        $SupressHostOutput
+    )
+
+    $ErrorActionPreference = 'Continue'
+
+    # Log how we are running docker for troubleshooting issues
+    Write-Verbose "Running docker $command $params" -Verbose
+    if($SupressHostOutput.IsPresent)
+    {
+        $result = &'docker' $command $params 2>&1
+    }
+    else 
+    {
+        &'docker' $command $params 2>&1 | Tee-Object -Variable result -ErrorAction SilentlyContinue | Out-String -Stream -ErrorAction SilentlyContinue | Write-Host -ErrorAction SilentlyContinue
+    }
+
+    $dockerExitCode = $LASTEXITCODE
+    if($PassThru.IsPresent)
+    {
+        Write-Verbose "passing through docker result$($result.length)..." -Verbose
+        return $result
+    }
+    elseif($dockerExitCode -ne 0 -and $FailureAction -eq 'error')
+    {
+        Write-Error "docker $command failed with: $result" -ErrorAction Stop
+        return $false
+    }
+    elseif($dockerExitCode -ne 0 -and $FailureAction -eq 'warning')
+    {
+        Write-Warning"docker $command failed with: $result"
+        return $false
+    }
+    elseif($dockerExitCode -ne 0)
+    {
+        return $false
+    }
+    
+    return $true
+}
+
+function Get-LinuxContainers 
+{
+    return @(
```

Comment:
since these are pattern driven, would this avoid typos?
```
foreach($os in "centos7","opensuse42.1","ubuntu14.04","ubuntu16.04") {
@{ Name = ${os}; Path = "${PsScriptRoot}/../release/${os}" }
}
```